sensor/common/virtualmachine/component.go

```diff
@@ -22,6 +24,6 @@ func NewComponent() Component {
 		centralReady:    concurrency.NewSignal(),
 		lock:            &sync.Mutex{},
 		stopper:         concurrency.NewStopper(),
-		virtualMachines: make(chan *storage.VirtualMachine),
+		virtualMachines: make(chan *storage.VirtualMachine, virtualMachineBufferedChannelSize),
```

Comment:
(nice to have) I would suggest to start with much smaller buffer - sth between 1 and 5 - or even without it to make it easier to discover any potential deadlocks in the tests. Once we are sure that there are no issues, we can increase it based on observations (using metrics for example).
Having it set to 100 would make it difficult to craft a unit or integration test that would fill this buffer.

Reply:
I put the buffer size to 1 for now, but I think we should bump it again before release.